helix-lsp/src/client.rs

```diff
@@ -145,6 +151,93 @@ impl Client {
             .expect("language server not yet initialized!")
     }
 
+    #[inline] // TODO inline?
```

Comment:
I think it's big on the surface (in code), but small when compiled (but that's just a feeling, this should probably be tested), it's also only used once currently, and in a separate crate, which AFAIK is a reason why this will otherwise not be inlined, even if it would makes sense performance-wise for the compiler. That's why I have for now added `#[inline]` and a `TODO` there, because I'm not completely sure.

Reply:
I do think that this function probably has quite a bit of assembly (although not nearly as much as it looks). Matches rarely compile down to the ideal assembly we imagine. Adding `#[inline]` is probably fine regardless. `#[inline]` only allows inlining but doesn't force it so worst case LLVM decides it's not worth inlining I think we can leave it here.